pkg/webhook/webhook.go

```diff
@@ -175,3 +192,102 @@ func (c *Webhook) validateHandlerFunc(review *admission_v1beta1.AdmissionReview)
 		Allowed: true,
 	}
 }
+
+func (c *Webhook) validateRelease(request *admission_v1beta1.AdmissionRequest, release shipper.Release) error {
+	var err error
+	overrideRBs := rolloutblock.NewOverride(release.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+	err = c.validateOverrideRolloutBlockAnnotation(overrideRBs, release.Namespace)
+	if err != nil {
+		return err
+	}
+
+	switch request.Operation {
+	case kubeclient.Create:
+		err = c.processRolloutBlocks(release.Namespace, overrideRBs)
+	case kubeclient.Update:
+		var oldRelease shipper.Release
+		err = json.Unmarshal(request.OldObject.Raw, &oldRelease)
+		if err != nil {
+			return nil
+		}
+		newOverrides := rolloutblock.NewOverride(release.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+		oldOverrides := rolloutblock.NewOverride(oldRelease.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+		subtractedOverrides := oldOverrides.Diff(newOverrides)
+		if len(subtractedOverrides) == 0 {
+			// no overrides were removed from annotation. validating release:
+			err = c.processRolloutBlocks(release.Namespace, overrideRBs)
+		} else {
+			// rolloutblock override were changed, making sure spec was not changed
+			if !reflect.DeepEqual(release.Spec, oldRelease.Spec) {
+				err = fmt.Errorf("releases Spec was changed %v", release)
+			}
+		}
+	default:
+	}
+
+	return err
+}
+
+func (c *Webhook) validateApplication(request *admission_v1beta1.AdmissionRequest, application shipper.Application) error {
+	var err error
+	overrideRBs := rolloutblock.NewOverride(application.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+	err = c.validateOverrideRolloutBlockAnnotation(overrideRBs, application.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if request.Operation == kubeclient.Create {
+		err = c.processRolloutBlocks(application.Namespace, overrideRBs)
+	}
+
+	return err
+}
+
+func (c *Webhook) processRolloutBlocks(namespace string, overrideRBs rolloutblock.Override) error {
+	existingRBs := rolloutblock.NewOverrideFromRolloutBlocks(c.existingRolloutBlocks(namespace))
```

Comment:
This right here makes me think `NewOverrideFromRolloutBlocks` doesn't create an override list, but just a list of rollout blocks. Maybe we should rename this.